src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -96,6 +96,42 @@ void IMergeTreeDataPart::MinMaxIndex::load(const MergeTreeData & data, const Par
     initialized = true;
 }
 
+Block IMergeTreeDataPart::MinMaxIndex::getBlock(const MergeTreeData & data) {
```

Comment:
In order to calculate the new MinMax partition keys, a `Block` with current MinMax values is required.